### PR TITLE
Add the SongsTable and other minor changes.

### PR DESCRIPTION
**Pull request by snmlabs**

### CODE REVIEW

File: app/src/main/java/com/bhakti/db/tables/SongsTable.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  // SQL statements
+  private static final String CREATE_SONGS_TABLE_SQL =
+      String.format("CREATE TABLE %s " +
+              "( %s INTEGER PRIMARY KEY, " +
+              " %s TEXT, " +
+              " %s TEXT, " +
+              " %s INTEGER, " +
+              " %s INTEGER, " +
+              " %s INTEGER, " +
```
Comment (sasi306) [inline app/src/main/java/com/bhakti/db/tables/SongsTable.java]: nit: not using %s will be more readable in this specific case.

"CREATE TABLE " +
SongEntry.TABLE_NAME +
" ( " +
SongEntry._ID +
"INTEGER PRIMARY KEY, "
.
.


It is tough to map which %s belongs to which argument and we can easily mess this up

File: app/src/main/java/com/bhakti/db/tables/SongsTable.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  @Override
+  public void onCreate(SQLiteDatabase sqLiteDatabase) {
+    Log.d(TAG, String.format("onCreate: Creating %s table",
+        SongEntry.TABLE_NAME));
+    sqLiteDatabase.execSQL(CREATE_SONGS_TABLE_SQL);
+  }
+
+  @Override
+  public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
```
Comment (sasi306) [inline app/src/main/java/com/bhakti/db/tables/SongsTable.java]: Can can add comments about when this function will be called. May be it is very obvious but i am still not sure of all the things yet. 

This does look like a costly calculation though.

File: app/src/main/java/com/bhakti/db/tables/SongsTable.java (hunk 1 above)
Comment (snmlabs) [reply app/src/main/java/com/bhakti/db/tables/SongsTable.java]: Done.

File: app/src/main/java/com/bhakti/db/tables/SongsTable.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          SongEntry.COL_DURATION_SECS_NAME,
+          SongEntry.COL_YEAR_NAME,
+          SongEntry.COL_TRACK_NUMBER_NAME,
+          SongEntry.COL_CLOUD_SONG_ABS_PATH_NAME,
+          SongEntry.COL_CLOUD_IMAGE_ABS_PATH_NAME,
+          SongEntry.COL_DEVICE_SONG_ABS_PATH_NAME,
+          SongEntry.COL_DEVICE_IMAGE_ABS_PATH_NAME,
+          SongEntry.COL_DISABLED_NAME);
+
+  private static final String DELETE_ENTRIES_SQL =
```
[the remark this answers is not in the document]
Comment (snmlabs) [reply app/src/main/java/com/bhakti/db/tables/SongsTable.java]: Done

File: app/src/main/java/com/bhakti/db/tables/SongsTable.java (hunk 2 above)
Comment (snmlabs) [reply app/src/main/java/com/bhakti/db/tables/SongsTable.java]: It is called when the database schema changes. That is, if there are any migrations. We are not planning to support any of this now.

Comment (sasi306) [reply app/src/main/java/com/bhakti/db/tables/SongsTable.java]: cool.

**Pull request merged**